### PR TITLE
add LiquidTotalIssuance

### PR DESCRIPTION
**Pull request by xlc**
Some of the tokens are out of circulation and simply cannot participate governance (e.g. the one locked for crowdloan). It will be unfair to include those into consideration for majority voting calculation.

This allow the runtime to configure what's the right number to be used.

Let me know if this is the right way and then I will do the companion PR.

### CODE REVIEW

File: frame/balances/src/lib.rs [hunk 1]
```diff
@@ -2189,3 +2189,11 @@ where
 		Self::update_locks(who, &locks[..]);
 	}
 }
+
+pub struct BalancesTotalIssuance<T, I = ()>(sp_std::marker::PhantomData<(T, I)>);
```
Comment (xlc) [inline frame/balances/src/lib.rs:2193]: `TotalIssuance` name is taken by the storage and can't think a better name. Maybe we should simply implement `Get` for `StorageValue`?

File: frame/democracy/src/lib.rs [hunk 2]
```diff
@@ -368,6 +368,10 @@ pub mod pallet {
 		/// The maximum number of public proposals that can exist at any time.
 		#[pallet::constant]
 		type MaxProposals: Get<u32>;
+
+		/// The maximumal possible amount of tokens that can participate in the governance.
+		/// Note: This is not a constant.
```
Comment (kianenigma) [inline frame/democracy/src/lib.rs:373]:
```suggestion
		/// 
		/// Note: This is not a constant.
```

**Review by kianenigma — COMMENTED** [review]
How do you want this to be implemented in your chain or elsewhere? 

I think the business of knowing the total issuance is a concern for the `T::Currency` correctly. Maybe you can add a new function to the `Currency` trait based on what you want to achieve?

File: frame/balances/src/lib.rs (hunk 1 above)
Comment (kianenigma) [reply frame/balances/src/lib.rs:2193]: >  Maybe we should simply implement Get for StorageValue?

`Get` is already ambiguous regarding being constant or not, I think what you did now is better. 

We are thinking of deprecating `Get` types as well.

**Comment by xlc** [comment]
For Acala, we will simply deduct the balances from a few known accounts (e.g. treasury). It won't reflect the real value but still a more suitable value.

For Polkadot, we could track number of DOT locked for crowdloan and deduct them, and deduct treasury holding as well.

Yeah add `liquid_total_issuance` is another solution. I don't really have a preference.

**Comment by xlc** [comment]
Apparently this is fixed for the new governance pallet so closing

https://github.com/paritytech/substrate/blob/32a4fe01f110a755bf22eb8f803cdfd7052d4f8b/frame/conviction-voting/src/lib.rs#L108-L112

**Pull request closed without merging**